HW3/P3/sum.cl

```diff
@@ -5,11 +5,15 @@ __kernel void sum_coalesced(__global float* x,
 {
     float sum = 0;
     size_t local_id = get_local_id(0);
+    size_t global_id = get_global_id(0);
+    uint gs = get_local_size(0);
 
     // thread i (i.e., with i = get_global_id()) should add x[i],
     // x[i + get_global_size()], ... up to N-1, and store in sum.
-    for (;;) { // YOUR CODE HERE
-        ; // YOUR CODE HERE 
+    for (uint c = global_id; c < N; c = c + get_global_size(0)) { 
+              if( global_id < N ){
```

Comment:
Not sure if/why this is necessary.
If initially c = global_id, and the for loop condition is c<N, then why would a thread with a global_id >= N enter the loop execution?